boards: PHYTEC: remove SoC prefix from board name

jonas-rem opened the pull request:
For easier recognition, PHYTEC boards have been prefixed with the SoC name. As the new hardware model includes the SoC and cpu, prefixing is not needed anymore. All PHYTEC eval boards have an individual and unique name and can be found easily via this name.

dnltz commented on boards/phytec/phyboard_polis/doc/index.rst:
```diff
@@ -1,4 +1,4 @@
-.. _mimx8mm_phyboard_polis:
+.. _phyboard_polis:
 
 PhyBOARD Polis (NXP i.MX8M Mini)
```
Can we fix both titles in this PR as well?

Should be called `phyBOARD-Polis (NXP i.MX8M Mini)` or `phyBOARD-Polis i.MX8M Mini`

jonas-rem replied:
Just fixed all occasions of the name and corrected it for both boards. Should now be aligned over all PHYTEC boards.

dnltz commented on boards/phytec/phyboard_pollux/doc/index.rst:
```diff
@@ -1,4 +1,4 @@
-.. _mimx8mp_phyboard_pollux:
+.. _phyboard_pollux:
 
 PhyBOARD Pollux (NXP i.MX8M Plus)
```
Same here.

nordicjm reviewed (changes requested):
`boards/deprecated.cmake` needs update

dnltz commented:
@jonas-rem  Here is a reference for `boards/deprecated.cmake`: https://github.com/zephyrproject-rtos/zephyr/blob/9f60075d14a283602a3580b6b97dd4686b53a28e/boards/deprecated.cmake#L49

kartben reviewed (changes requested):
A welcome but breaking change so please mention it in the migration guide.
Also add redirect entries in doc/scripts/_redirects.py

Edit: possibly non breaking as others mentioned by using deprecated board name. Still needs a mention in the release notes to give people a heads-up though

jonas-rem commented:
Thanks for the hints @nordicjm, @kartben, @dnltz!

Added an entry to `deprecated.yaml`, `redirect.py` and `release-notes-4.0.rst `

kartben commented on boards/phytec/phyboard_polis/doc/index.rst:
```diff
@@ -1,7 +1,7 @@
 
-PhyBOARD Polis (NXP i.MX8M Mini)
-################################
+phyBOARD-Polis i.MX8M Mini
```
One last request from me: please set the new names in the new `full_name` property of the respective `board.yml` files. Thanks!

dnltz replied:
Ah nice, I will update the other boards soon too.

kartben reviewed (dismissed):
looking good :) https://builds.zephyrproject.io/zephyr/pr/79575/docs/boards/index.html#vendor=phytec

(don't mind the bug in the vendor select box -- it will be fixed with #79695)

dnltz reviewed (dismissed):
Thanks for updating those boards :)

dnltz commented:
Assigned @carlescufi since he's the code owner.

nordicjm commented on boards/deprecated.cmake:
```diff
@@ -916,3 +916,6 @@ set(yd_esp32_DEPRECATED
 set(mimx8mp_phyboard_pollux/mimx8ml8/m7_DEPRECATED
     phyboard_pollux/mimx8ml8/m7
 )
+set(mimx8mm_phyboard_polis/mimx8mm6/m4_DEPRECATED
```
line 388 needs updating too

jonas-rem replied:
Oh, right! The old deprecation that has been introduced when HWMv2 came must directly point to the most recent board. Will change.

Changed also l.368 (Polis).

nordicjm commented on boards/phytec/phyboard_polis/doc/index.rst:
```diff
@@ -70,8 +70,8 @@ More information about the board can be found at the
 
-The Zephyr ``mimx8mm_phyboard_polis/mimx8mm6/m4`` board configuration supports the following
-hardware features:
+The Zephyr ``phyboard_polis/mimx8mm6/m4`` board configuration supports the
```
*board target

jonas-rem replied:
fixed

jonas-rem commented:
Adressed the last two comments from @nordicjm.

jonas-rem commented:
I had to rebase on current main because of merge conflicts in the release notes file. There are no code changes introduced.

@kartben @nordicjm @dnltz @carlescufi could you please re-approve?

MaureenHelm commented:
Needs rebase

Commit:
boards: PHYTEC: phyboard_pollux: remove mimx8mp prefix

For easier recognition PHYTEC boards have been prefixed with the SoC name.
As the new hardware model includes the SoC and cpu, this prefixing is not
needed anymore. All PHYTEC eval boards have an individual and unique name
and can be found easily via this name.

Signed-off-by: Jonas Remmert <j.remmert@phytec.de>

jonas-rem commented:
Rebased on current main

Commit:
boards: PHYTEC: phyboard_polis: remove mimx8mm prefix

For easier recognition PHYTEC boards have been prefixed with the SoC name.
As the new hardware model includes the SoC and cpu, this prefixing is not
needed anymore. All PHYTEC eval boards have an individual and unique name
and can be found easily via this name.

Signed-off-by: Jonas Remmert <j.remmert@phytec.de>
